app/controllers/change_requests_controller.rb

```diff
@@ -256,14 +265,16 @@ def persist_change(change_request)
     elsif change_request.is_a? AddressChangeRequest
       puts "AddressChangeRequest"
       address = Address.find(change_request.object_id)
-      
-      a = geocode_address address
-      unless a.nil?
-        field_change_hash["latitude"] = a.latitude
-        field_change_hash["longitude"] = a.longitude 
+      if change_request.remove?
+        address.delete
```

Comment:
Generally, you'll want to call `#destroy`, not `#delete`. The difference is that `#destroy` will also delete any child models and call any other custom callback methods. See https://stackoverflow.com/a/22757533 for a slightly more detailed description and links to the official documentation.

Here, I believe the difference will become apparent with the Addresses–Services join table, where if you merely try to `#delete` an Address that has a relationship to a Service, then it will (probably?) cause a SQL constraints failure, because it won't try to destroy the Address–Service relationship, but the Address-Service relationship will have a foreign key to the Address you are trying to delete. I believe that `#destroy` will delete both the Address and its specific row in the Address-Service table, which will not hit an error.

It's been many years since I've seriously written Rails code like this, so there might be some mistakes in my description above, but I believe high level ideas are still correct.

Reply:
was not aware of this discrepancy!